components/brave_wallet/browser/zcash/zcash_serializer_unittest.cc

```diff
@@ -0,0 +1,188 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/zcash/zcash_serializer.h"
+
+#include <algorithm>
+#include <memory>
+#include <utility>
+
+#include "brave/components/brave_wallet/browser/zcash/zcash_transaction.h"
+#include "brave/components/brave_wallet/common/hex_utils.h"
+#include "brave/components/brave_wallet/common/zcash_utils.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+TEST(ZCashSerializerTest, HashPrevouts) {
```

Comment:
are these standardized test vectors listed somewhere or just tests to make sure code doesn't break if we refactor in the future?

Reply:
There are vectors, but they mostly depend on shielded parts too, so we can't just borrow them at this moment.
I've added a couple of tests based on real transactions here: ZCashSerializerTest::TxId_TransparentOnly and 
ZCashWalletServiceUnitTest, SignAndPostTransaction.
Tests for Hash* are just making easier to find which part is actually broken.

Reply:
We can likely generate these from here: https://github.com/zcash/zcash-test-vectors

Reply:
As i see there is no intermediate data like sub-digests in such vectors.
I think we can just reuse published vectors when shielded will be implemented for top-level things like tx id and sig digest.